esx_service/vmdk_ops.py

```diff
@@ -1786,11 +1826,16 @@ def handleVmciRequests(port):
     cartel = c_int32()
     vmci_grab_listening_socket(port)
 
-
     while True:
+        # Check the STOP flag - exit the loop if the flag is set
```

Comment:
I agree it's not really required, since we already have a check after the op is received. But this won't prevent client from seeing proper error messages, since we don't even try to get the ops at all after breaking the loop. Anyway, I will remove this check.

Reply:
After debugging this further - it proves that we do need this check. Here's what I observed:

1. After sending the SIGTERM signal, the blocking vmci_get_one_op() call will return with an error:
```
09/26/17 21:03:39 8172365 [MainThread] [DEBUG  ] lib.vmci_get_one_op returns -1, buffer 'b'{"cmd":"list","details":{"Name":""},"version":"2"}''
09/26/17 21:03:39 8172365 [MainThread] [WARNING] vmci_get_one_op failed ret=-1: Invalid argument (errno=22) Retrying...
```
2. Above error will cause the workflow continue and again being blocked on the vmci_get_one_op() call.

So, if we don't have above check, the loop will never end - it's being blocked on vmci_get_one_op() for ever. Please note that we don't call exit(0) from main thread anymore, as the wait_ops_in_flight() is now being invoked from a child thread.